### PR TITLE
Fix parser issues for incorrect indexes for Edit and Delete commands

### LINKED ISSUE #183: [PE-D][Tester E] Wrong error message given when deleting using an index that is too large.
To replicate: Give the command "delete e/99999999999"

Expected: An 'invalid index' error message

Actual: A 'non-integer detected' error message

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: naz019/ped#8

### PR DESCRIPTION
**Pull request by chinhan99**

### CODE REVIEW

File: src/main/java/seedu/duke/common/ErrorMessages.java [hunk 1]
```diff
@@ -18,6 +18,7 @@ public enum ErrorMessages {
     ERROR_GLOBAL_INVALID_PERIOD("Type of period given is invalid, please check your input!"),
     ERROR_GLOBAL_INVALID_NUMBER("Number for period stats given is invalid, please check your input!"),
     ERROR_GLOBAL_NUMBER_NOT_NUMERIC("Non-integer detected, please enter a numerical integer!"),
+    ERROR_GLOBAL_NON_NUMERIC_INDEX("Non-numerical index detected, please enter a numerical index!"),
```
Comment (chydarren) [inline src/main/java/seedu/duke/common/ErrorMessages.java]: Changes look good, there's a clear distinction between requiring an integer amount and an integer index. Perhaps you would like to change to `ERROR_GLOBAL_INDEX_NOT_NUMERIC` for clarity?

File: src/main/java/seedu/duke/parser/ParameterParser.java [hunk 2]
```diff
@@ -473,17 +479,28 @@ public static LocalDate parseDateTag(String parameter) throws InputTransactionIn
      */
     public static int parseEntryTag(String parameter) throws MoolahException {
         int index;
+        //@@chinhan99
+        if (containAlphabet(parameter)) { // Checks if parameter contains alphabets
```
Comment (chydarren) [inline src/main/java/seedu/duke/parser/ParameterParser.java:483]: Wow! That's a stronger check to ensure that the index is strictly integer and no special symbols allowed.

File: src/main/java/seedu/duke/parser/ParameterParser.java [hunk 3]
```diff
@@ -685,4 +702,39 @@ public static boolean containSymbol(String parameter) {
...
+        if (parameter.startsWith(String.valueOf(minusSymbol))) {
+            while (hasMinusSymbol.find()) {
+                countOfMinusSymbol += 1;
+            }
+        } else { // immediately return false as parameter does not start with minus sign
+            return false;
+        }
+
+        // At this stage, parameter definitely starts with minus, and double check how many minus signs are in parameter
+        if (countOfMinusSymbol == 1) {
```
Comment (chydarren) [inline src/main/java/seedu/duke/parser/ParameterParser.java:738]: You may wish to give the number `1` some meaning using pre-defined constants to better describe the magic literal.

File: src/main/java/seedu/duke/common/ErrorMessages.java (hunk 1 above)
Comment (chinhan99) [reply src/main/java/seedu/duke/common/ErrorMessages.java]: Good point! Changed in the most recent commit.

**Pull request merged**